### PR TITLE
[Favourite] Implement the messages favourite feature

### PR DESCRIPTION
**Pull request by Yoshiin**
Resolve dinsic-pim/tchap-web#150

### CODE REVIEW

File: src/components/views/context_menus/MessageContextMenu.js [hunk 1]
```diff
@@ -355,8 +369,17 @@ module.exports = React.createClass({
                 </div>;
         }
 
+        let favouriteClasses = "mx_MessageContextMenu_field mx_MessageContextMenu_field_icon";
+        favouriteClasses += this.state.isFavourite ? " mx_MessageContextMenu_field_favourite" : " mx_MessageContextMenu_field_favourite_out";
+        const favoriteButton = (
```
Comment (giomfo) [inline src/components/views/context_menus/MessageContextMenu.js:374]: rename with "favouriteButton"

Comment (Yoshiin) [reply src/components/views/context_menus/MessageContextMenu.js:374]: Done.

File: src/TchapFavouriteManager.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "tags" : {
+                "m.favourite": {}
+            }
+        };
+
+        MatrixClientPeg.get().setRoomAccountData(room.roomId, "m.tagged_events", taggedEvents);
+    }
+
+    static getFavorite(room) {
+        return room.getAccountData("m.tagged_events");
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/TchapFavouriteManager.js:55]: Done.

File: src/TchapFavouriteManager.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        if (roomAccountData) {
+            favouriteEvents = roomAccountData.event.content.tags["m.favourite"];
+        }
+
+        delete favouriteEvents[eventId];
+
+        const taggedEvents = {
+            "tags" : {
+                "m.favourite": {}
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/TchapFavouriteManager.js:47]: Done.

File: src/TchapFavouriteManager.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            "tags" : {
+                "m.favourite": {
+                    ...oldFavEvents,
+                    [event.getId()]: {
+                        "origin_server_ts": event.getTs(),
+                        "tagged_at": Date.now()
+                    }
+                }
+            }
+        };
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/TchapFavouriteManager.js:28]: Fixed.

File: src/TchapFavouriteManager.js [hunk 5]
```diff
@@ -23,7 +40,8 @@ class TchapFavouriteManager {
                         "origin_server_ts": event.getTs(),
                         "tagged_at": Date.now()
                     }
```
Comment (giomfo) [inline src/TchapFavouriteManager.js:42]: Don't you think you should check if the event is not already favourite before adding it?

**Review by giomfo — APPROVED** [review]
approved, just one comment

**Review by giomfo — APPROVED** [review]

**Pull request merged**